File: net/Azure.Security.KeyVault.Secrets.PerfStress/GetSecretTest.cs
using Azure.Security.KeyVault.Secrets.PerfStress.Core;
using Azure.Test.PerfStress;
using System;
using System.Threading;
using System.Threading.Tasks;

namespace Azure.Security.KeyVault.Secrets.PerfStress
{
    public class GetSecretTest : SecretTest<PerfStressOptions>
    {
        private const string _secretName = "GetSecretName";
        private const string _secretValue = "GetSecretValue";

        public GetSecretTest(PerfStressOptions options) : base(options)
        {
        }

        public override async Task GlobalSetupAsync()
        {
            await base.GlobalSetupAsync();

            await SecretClient.SetSecretAsync(_secretName, _secretValue);
        }

        public override async Task GlobalCleanupAsync()
        {
            await SecretClient.StartDeleteSecretAsync(_secretName);

            await base.GlobalCleanupAsync();
        }

        public override void Run(CancellationToken cancellationToken)
        {
            var _ = SecretClient.GetSecret(_secretName).Value.Value;
        }

        public override async Task RunAsync(CancellationToken cancellationToken)
        {
            var _ = (await SecretClient.GetSecretAsync(_secretName)).Value.Value;
        }
    }
}

File: sdk/keyvault/azure-security-keyvault-perf/src/main/java/com/azure/security/keyvault/secrets/perf/GetSecretTest.java
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
package com.azure.security.keyvault.secrets.perf;

import com.azure.perf.test.core.PerfStressOptions;
import com.azure.security.keyvault.secrets.perf.core.SecretsTest;
import reactor.core.publisher.Mono;

import java.util.UUID;

public class GetSecretTest extends SecretsTest<PerfStressOptions> {
    private static final String SECRET_NAME = "getSecretPerfTest-" + UUID.randomUUID();

    public GetSecretTest(PerfStressOptions options) {
        super(options);
    }

    @Override
    public Mono<Void> globalSetupAsync() {
        return super.globalSetupAsync()
            .then(secretAsyncClient.setSecret(SECRET_NAME, SECRET_NAME))
            .then();
    }

    @Override
    public Mono<Void> globalCleanupAsync() {
        return deleteAndPurgeSecretsAsync(SECRET_NAME)
            .then(super.globalCleanupAsync());
    }

    @Override
    public void run() {
        secretClient.getSecret(SECRET_NAME);
    }

    @Override
    public Mono<Void> runAsync() {
        return secretAsyncClient.getSecret(SECRET_NAME).then();
    }
}
